### PR TITLE
Arakune

### PR DESCRIPTION
**Pull request by daniel-k-taylor**

### CODE REVIEW

File: data/card_definitions.json [hunk 1]
```diff
@@ -338,6 +338,458 @@
...
+			}
+		],
+		"boost": {
+			"boost_type": "immediate",
+			"force_cost": 1,
+			"cancel_cost": -1,
+			"display_name": "Forbidden Knowledge",
+			"effects": [
+				{
+					"timing": "immediate",
```
Comment (Mach131) [inline data/card_definitions.json:433]: Does this boost have the same potential ordering issues as Plague Knight's boost that looks at the top 3 cards of deck? honestly i have no idea how you'd reasonably get around it with 5 cards so I hope not but

File: data/decks/arakune.json [hunk 2]
```diff
@@ -0,0 +1,227 @@
+{
+	"id": "arakune",
+	"season": 5,
+	"display_name": "Arakune",
+	"cardback": "cardback_arakune.jpg",
+	"exceed_cost": 8,
+	"exceed_cost_reduced_by": "overdrive_count",
+	"has_sealed_area": true,
+	"sealed_area_is_facedown": true,
```
Comment (Mach131) [inline data/decks/arakune.json:9]: were we changing this to is_secret?

File: data/decks/arakune.json [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+		{
+			"timing": "on_reshuffle",
+			"condition": "manual_reshuffle",
+			"effect_type": "add_set_aside_card_to_deck",
+			"id": "arakune_ntoinfinity"
+		},
+		{
+			"timing": "on_strike_reveal",
+			"effect_type": "choose_discard",
+			"source": "overdrive",
```
Comment (Mach131) [inline data/decks/arakune.json:60]: can probably add character_effect flag in here somewhere safely

File: data/decks/arakune.json [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+				"amount": -8,
+				"amount2": 0,
+				"and": {
+					"effect_type": "powerup",
+					"amount": 2,
+					"and": {
+						"effect_type": "speedup",
+						"amount": 1,
+						"and": {
+							"NOTE": "Overdrive Effect",
```
Comment (Mach131) [inline data/decks/arakune.json:77]: oh I did not know that overdrive effects went off any time you discarded from overdrive

File: scenes/game/local_game.gd [hunk 5]
```diff
@@ -1826,24 +1911,39 @@ class Player:
...
 						strike_stat_boosts.remove_ex()
 					"dodge_at_range":
-						var dodge_range = str(strike_stat_boosts.dodge_at_range_min)
-						if strike_stat_boosts.dodge_at_range_min != strike_stat_boosts.dodge_at_range_max:
-							dodge_range += "-%s" % strike_stat_boosts.dodge_at_range_max
-						parent._append_log_full(Enums.LogType.LogType_Effect, self, "will no longer dodge attacks from range %s." % dodge_range)
-						strike_stat_boosts.dodge_at_range_min = -1
-						strike_stat_boosts.dodge_at_range_max = -1
-						strike_stat_boosts.dodge_at_range_from_buddy = false
+						if 'special_range' in effect:
```
Comment (Mach131) [inline scenes/game/local_game.gd:1914]: probably should check the value of the special range just for future clarity

File: scenes/game/local_game.gd [hunk 6]
```diff
@@ -2468,6 +2574,14 @@ func is_effect_condition_met(performing_player : Player, effect, local_condition
 		elif condition == "life_equals":
 			var amount = effect['condition_amount']
 			return performing_player.life == amount
+		elif condition == "discarded_matches_attack_speed":
+			var discarded_card_ids = effect['discarded_card_ids']
+			assert(discarded_card_ids.size() == 1)
+			var card = card_db.get_card(discarded_card_ids[0])
+			var speed_of_discarded = card.definition['speed']
+			var attack_card = active_strike.get_player_card(performing_player)
+			var printed_speed_of_attack = attack_card.definition['speed']
```
Comment (Mach131) [inline scenes/game/local_game.gd:2583]: this effect only cares about printed speeds right? (i.e. if you have light in play)

**Review by Mach131 — CHANGES_REQUESTED** [review]
arakune weird; mostly trusting in the test cases/rose's manual tests tho

File: data/decks/arakune.json (hunk 2 above)
Comment (daniel-k-taylor) [reply data/decks/arakune.json:9]: whoops, forgot about this

File: data/decks/arakune.json (hunk 4 above)
Comment (daniel-k-taylor) [reply data/decks/arakune.json:77]: arakune explicitly calls this on his card

File: scenes/game/local_game.gd (hunk 5 above)
Comment (daniel-k-taylor) [reply scenes/game/local_game.gd:1914]: just that it is set to something non- ""?
I can add that.

File: scenes/game/local_game.gd (hunk 6 above)
Comment (daniel-k-taylor) [reply scenes/game/local_game.gd:2583]: yes, I double checked the faq

File: data/card_definitions.json (hunk 1 above)
Comment (daniel-k-taylor) [reply data/card_definitions.json:433]: I think this is okay. and even if it isn't, too bad.

**Review by Mach131 — APPROVED** [review]

**Pull request merged**